WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m

```diff
@@ -73,29 +73,35 @@ - (XCElementSnapshot *)fb_lastSnapshot
     return @[];
   }
   NSArray<NSNumber *> *matchedUids = [snapshots valueForKey:FBStringify(XCUIElement, wdUID)];
+  NSMutableArray<XCUIElement *> *matchedElements = [NSMutableArray array];
+  if ([matchedUids containsObject:@(self.wdUID)]) {
+    if (1 == snapshots.count) {
+      return @[self];
```

Comment:
_snapshots_ array defines how many elements should be returned. The situation where single snapshot matched to multiple elements is not possible. This means if we only need to match one snapshot and the uid of this snapshot matched to the uid of self element then the result should be single self element